### PR TITLE
Frontend dashboard

### PR DESCRIPTION
**Pull request by ALI-KHALED-13**

### CODE REVIEW

File: client/src/components/BarChart/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  BarElement,
+  Title,
+  Tooltip,
+);
+
+interface BarChartProps {
+  data: ChartData<"bar", (number | [number, number] | null)[], string>/*{
```
Comment (Elena-Erganyan) [inline client/src/components/BarChart/index.tsx:22]: To get rid of the TS error here add the following:
`import { ChartData } from "chart.js";`

**Review by Elena-Erganyan — CHANGES_REQUESTED** [review]
Great work Ali 👏 Really liked it, so many things were taken in account
It looks nice overall, some small comments UI wise are coming soon
Left some comments regarding your code, in general it's good :)

**Comment by Elena-Erganyan** [comment]
UI comments:
1) Common:
- there should be `max-width: 111rem` as on the other pages
- for consistency purposes would be nice to make border-radius for both blocks (`UserProfile` and `StyledContributionsArea`) 1rem
- a bit more of padding would look better (1.5 - 2rem for each block)
2) `UserProfile` part:
- icons of delete and upload image buttons are not centered vertically
- in my opinion it would be better to put email as the 3rd property (logically and because it can't be edited for now)
3) `StyledContributionsArea` part:
- to avoid the messiness in styling it's better to take colors for tabs from `GlobalStyles`
- also I think `Overview` tab also should have some highlighting color
- products are not clickable (there is no link to the product page there)
- feedback look like this:
Screenshot_1
so we can't see its borders
`box-shadow: 0 0 10px 1px #CFD7FF` would do it I guess :)
- on some mobile screens we have this (example: IPhone X)
Screenshot_2
Having a scrollbar is okay, the problem is that the left tab is not fully shown
- there should be some gap between the products (feedback, comments)
Screenshot_3

File: client/src/redux/api/productsApi.ts [hunk 2]
```diff
@@ -7,6 +7,11 @@ export const productsApi = api.injectEndpoints({
       keepUnusedDataFor: 600,
       providesTags: [{type: 'Products', id: 'List'}],
     }),
+    getProductsByField: builder.query<IProduct[], {key: string, value: string}>({
+      query: ({key, value}) => `products?${key}=${value}`,
+      keepUnusedDataFor: 600,
+      providesTags: (_result, _error, arg)=> [{type: 'Products', id: `List_${arg.key}=${arg.value}`}],
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply client/src/redux/api/productsApi.ts:13]: thanks for pointing it out, I totally missed changing the invalidation part.. so imagine this scenario:
user going to dashboard - product tab, then adding a product and going back to that tab.. the data in that tab must be refetched (invalidated) to show the new product added

File: client/src/redux/api/feedbackApi.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ const feedbackApi = api.injectEndpoints({
     getFeedbackList: builder.query<IFeedback[], {key: string, value: string}>({
       query: ({key, value}) => `feedback?${key}=${value}`,
       keepUnusedDataFor: 300,
-      providesTags: [{ type: 'Feedback', id: 'LIST' }],
+      providesTags: (_result, _error, arg)=> [{ type: 'Feedback', id: 'LIST' }],
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply client/src/redux/api/feedbackApi.ts:8]: wanted to talk about it

File: client/src/pages/Dashboard/index.tsx [hunk 4]
```diff
@@ -1,15 +1,144 @@
...
+              chartTitle="Your activity overview"
+            />
+          </div>
+        )}
+        {activeQueryState && (
+          activeQueryState.isLoading || activeQueryState.isFetching?
+          <Loader text="Loading.." isBlock blockHeight="50vh"/>
+          :
+          activeQueryState.error?
+          <div>{getErrorMessage(activeQueryState.error) as string}</div>
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply client/src/pages/Dashboard/index.tsx:134]: I think displaying it here would be better

File: client/src/components/UserProfile/index.tsx [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    }
+    if (user.pfp?.url) {
+      deletePFP();
+    } else {
+      setErrorMessage("you already have no custom profile picture to delete")
+    }    
+  }
+
+  const handleFileUpload =(ev:ChangeEvent)=> {
+    const fileInput = ev.target as HTMLInputElement;
```
[the remark this answers is not in the document]
Comment (ALI-KHALED-13) [reply client/src/components/UserProfile/index.tsx:89]: thanks :)

**Review by Elena-Erganyan — APPROVED** [review]
Well done 😃

**Pull request merged**